projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -282,8 +299,155 @@ export class IgxDatePickerComponent implements ControlValueAccessor, EditorProvi
         this._specialDates = value;
     }
 
+    @Input()
+    public get modalOverlaySettings(): OverlaySettings {
+        return this._modalOverlay;
+    }
+
+    public set modalOverlaySettings(value: OverlaySettings) {
+        this._modalOverlay = value;
+    }
+
+    @Input()
+    public get dropDownOverlaySettings(): OverlaySettings {
+        return this._dropDownOverlay;
+    }
+
+    public set dropDownOverlaySettings(value: OverlaySettings) {
+        this._dropDownOverlay = value;
+    }
+
     /**
-     *An @Input proeprty that sets the orientation of the `IgxDatePickerComponent` header.
+     *Returns the formatted date when `IgxDatePickerComponent` is readonly.
+     *```typescript
+     *@ViewChild("MyDatePicker")
+     *public datePicker: IgxDatePickerComponent;
+     *public selection(event){
+     *    let selectedDate = this.datePicker.displayData;
+     *    alert(selectedDate);
+     *}
+     *```
+     *```html
+     *<igx-date-picker #MyDatePicker (onSelection)="selection()" todayButtonLabel="today"></igx-date-picker>
+     *```
+     */
+    public get displayData(): string {
+        if (this._value) {
+            return this._customFormatChecker(this.formatter, this._value);
+        }
+    }
+
+    /**
+     hidden
+     */
+    public get transformedDate(): string {
+        if (this._value) {
+            if (this._isInEditMode) {
+                this._transformedDate = this._getEditorDate(this._value);
+            } else {
+                this._transformedDate = this._getDisplayDate(this._value);
+            }
+
+            this.isEmpty = false;
+        }
+        return this._transformedDate;
+    }
+
+    public set transformedDate(value) {
+        this._transformedDate = value;
+    }
+
+    constructor(@Inject(IgxOverlayService) private _overlayService: IgxOverlayService,
+        private cdr: ChangeDetectorRef) { }
+
+    /**
+     * Gets the input group template.
+     * ```typescript
+     * let template = this.template();
+     * ```
+     * @memberof IgxDatePickerComponent
+     */
+    get template(): TemplateRef<any> {
+        if (this.datePickerTemplateDirective) {
+            return this.datePickerTemplateDirective.template;
+        }
+        return (this.mode === DatePickerInteractionMode.READONLY) ? this.readOnlyDatePickerTemplate : this.editableDatePickerTemplate;
+    }
+
+    /**
+     * Gets the context passed to the input group template.
+     * @memberof IgxDatePickerComponent
+     */
+    get context() {
+        return {
+            value: this.value,
+            displayData: this.displayData,
```

Comment:
`displayDate` if it's not a breaking change.

Reply:
it will be a breaking change.